File: C++ Programming/Competitive Programming/Coding Challenges 10/program3.cpp
// 3.Write a program to find factorial of given number.
// Input : 5
// Output : 120 (5 * 4 * 3 * 2 * 1)
// Input : -5
// Output : 120 (5 * 4 * 3 * 2 * 1)
// Input : 4
// Output : 24 (4 * 3 * 2 * 1)

#include <iostream>
using namespace std;

int Factorial(int iNo)
{
    int iFact = 1;

    if (iNo < 0)
    {
        iNo = -iNo;
    }

    for (int i = 1; i <= iNo; i++)
    {
        iFact = iFact * i;
    }

    return iFact;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout << "Enter number: ";
    cin >> iValue;

    if (iValue < 0)
    {
        iValue = -iValue;
    }

    iRet = Factorial(iValue);

    cout << "Factorial of number is : " << iRet << " (";

    for (int i = iValue; i >= 1; i--)
    {
        cout << i;

        if (i != 1)
        {
            cout << " * ";
        }
    }

    cout << ")";

    return 0;
}

File: Java Programming/Competitive Programming/Coding Challenges 13/program3.java
// 3.Write a program to find even factorial of given number.
// Input : 5
// Output : 8 (4 * 2)
// Input : -5
// Output : 8 (4 * 2)
// Input : 10
// Output : 3840 (10 * 8 * 6 * 4 * 2)

import java.util.Scanner;

class program3
{
    static int EvenFactorial(int iNo)
    {
        if (iNo < 0)
        {
            iNo = -iNo;
        }

        if (iNo <= 1)
        {
            return 1;
        }

        if (iNo % 2 != 0)
        {
            iNo--;
        }

        return iNo * EvenFactorial(iNo - 2);
    }

    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;
        int iRet = 0;

        System.out.print("Enter number");
        iValue = sobj.nextInt();

        iRet = EvenFactorial(iValue);

        System.out.print("Even Factorial of number is " + iRet);

        sobj.close();
    }
}